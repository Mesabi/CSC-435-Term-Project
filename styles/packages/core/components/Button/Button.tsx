import styled, { th } from '@xstyled/styled-components';

const Button = styled.button`
  background-color: material;
  padding: 7 20 5;
  border: none;
  color: materialText;

  font-size: 12px;

  box-shadow: inset 1px 1px 0px 1px ${th('colors.borderLightest')},
    inset 0 0 0 1px ${th('colors.borderDark')},
    1px 1px 0 0px ${th('colors.borderDarkest')};

  &:disabled {
    color: materialTextDisabled;
  }

  &:focus {
    outline: ${th('space.1')}px dotted ${th('colors.materialText')};
    outline-offset: -${th('space.5')}px;

    box-shadow: inset 1px 1px 0px 1px ${th('colors.borderLightest')},
      inset -0.5px -0.5px 0px 1px ${th('colors.borderDark')},
      1px 1px 0 1px ${th('colors.borderDarkest')};
  }

  &:active {
    padding: 8 20 4;

    outline: ${th('space.1')}px dotted ${th('colors.borderDarkest')};
    outline-offset: -${th('space.5')}px;

    box-shadow: inset 0 0 0 1px ${th('colors.borderDark')},
      0 0 0 1px ${th('colors.borderDarkest')};
  }
`;

Button.defaultProps = {
  children: 'Ok',
};

export default Button;
