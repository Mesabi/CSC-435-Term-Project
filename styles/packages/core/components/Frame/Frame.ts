import styled, { system, SystemProps } from '@xstyled/styled-components';

export interface FrameProps extends SystemProps {}

const Frame = styled.div<FrameProps>(system);

Frame.defaultProps = {
  w: 'auto',
  h: 'auto',
  bg: 'material',
  boxShadow: 'out',
};

export default Frame;
