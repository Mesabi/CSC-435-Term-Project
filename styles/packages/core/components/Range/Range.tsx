import styled, { css, th } from '@xstyled/styled-components';

const trackStyle = css`
  width: 100%;
  height: 4px;

  background-color: borderDarkest;

  cursor: pointer;

  border-top: 1;
  border-top-color: borderLight;
  border-right: 2;
  border-right-color: borderLightest;
  border-bottom: 1;
  border-bottom-color: borderLighter;
  border-left: 2;
  border-left-color: borderLightest;

  box-shadow: 0px 1px 0px 0px ${th('colors.borderLightest')};
`;

const thumbStyle = css`
  width: 12px;
  height: 20px;

  background-color: material;

  cursor: pointer;

  margin-top: -8;

  border-top: 1;
  border-top-color: borderLightest;
  border-right: 1;
  border-right-color: borderDarkest;
  border-bottom: 1;
  border-bottom-color: borderDarkest;
  border-left: 1;
  border-left-color: borderLightest;

  box-shadow: inset 0px -1px 0px ${th('colors.borderDark')},
    inset -1px 0px 0px ${th('colors.borderDark')},
    inset 0px 1px 0px ${th('colors.borderLighter')},
    inset 1px 0px 0px ${th('colors.borderLighter')};

  -webkit-appearance: none;
`;

const Range = styled.input.attrs({
  type: 'range',
})`
  -webkit-appearance: none;
  width: 100%;

  &:hover,
  &:focus,
  &:active {
    outline: none;
  }

  &::-webkit-slider-runnable-track {
    ${trackStyle}
  }

  &::-webkit-slider-thumb {
    ${thumbStyle}
  }

  &:focus::-webkit-slider-runnable-track {
    background-color: borderDarkest;
  }

  &::-moz-range-track {
    ${trackStyle}
  }

  &::-moz-range-thumb {
    ${thumbStyle}
  }
`;

export default Range;
