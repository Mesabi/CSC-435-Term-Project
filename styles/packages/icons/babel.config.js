module.exports = {
  extends: '../../babel.config.js',
  ignore: ['**/*.d.ts'],
};
