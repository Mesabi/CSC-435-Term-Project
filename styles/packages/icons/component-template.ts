// @ts-check

import { IconData } from './processIcons';

export const getIconDataName = (name: string): string => {
  const lowercase = name.charAt(0).toLowerCase() + name.slice(1);
  return `${lowercase}Data`;
};

export const iconComponentTemplate = (
  name: string,
  images: IconData[],
): string => {
  const iconDataVariableName = getIconDataName(name);
  const componentName = name;
  const componentPropsName = `${componentName}Props`;
  return `import React from 'react';
import { IconProps } from '../iconType';
import { BaseIcon } from '../internal';
${images
  .map(image => `import ${image.id} from '${image.reactImportPath}';`)
  .join('\n')}

export const ${iconDataVariableName} = {${images
    .map(
      image => `
  '${image.variant}': {
    imageSrc: ${image.id} as string,
    width: ${image.width},
    height: ${image.height},
  },`,
    )
    .join('\n')}
};

export interface ${componentPropsName} extends IconProps {
  /**
   * Icon variant to use.
   * also provides default styling with the correct height and width
   **/
  variant?: ${images.map(image => `'${image.variant}'`).join(' | ')};
}

export const ${componentName}: React.FC<${componentPropsName}> = ({
  variant = '${images[0].variant}',
  ...rest
}) => {
  const image = ${iconDataVariableName}[variant];

  return (
    <BaseIcon
      width={image.width}
      height={image.height}
      src={image.imageSrc}
      {...rest}
    />
  );
};
`;
};
