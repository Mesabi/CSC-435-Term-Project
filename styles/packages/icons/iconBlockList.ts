import { IconData } from './processIcons';

/** This list includes all icons seen as not including any visual output */
const iconBlockedIds = [
  'Access218_16x16_4',
  'Access219_16x16_4',
  'Access220_16x16_4',
  'Access221_16x16_4',
  'Access222_16x16_4',
  'Access223_16x16_4',
  'Access224_16x16_4',
  'Access225_16x16_4',
  'Access226_16x16_4',
  'Access227_16x16_4',
  'Access228_16x16_4',
  'Access229_16x16_4',
  'Access230_16x16_4',
  'Appwiz1501_16x16_4',
  'Appwiz1502_16x16_4',
  'Awfext326050_32x32_4',
  'Awfxcg321302_16x16_4',
  'Awfxcg321303_16x16_4',
  'Awfxcg321305_16x16_4',
  'Dial16x16_4',
  'Explorer104_32x32_4',
  'Filefind3_16x16_4',
  'Filepick_16x16_4',
  'Filexfer129_16x16_4',
  'Filexfer130_16x16_4',
  'Gcdef108_32x32_4',
  'Gcdef109_32x32_4',
  'Gcdef110_32x32_4',
  'Gcdef111_32x32_4',
  'Gcdef112_32x32_4',
  'Gcdef113_32x32_4',
  'Gcdef114_32x32_4',
  'Gcdef115_32x32_4',
  'Gcdef116_32x32_4',
  'Gcdef117_32x32_4',
  'Gcdef122_32x32_4',
  'Gcdef124_32x32_4',
  'Lights101_32x32_4',
  'Lights102_32x32_4',
  'Lights103_32x32_4',
  'Mshtml32539_16x16_4',
  'Mshtml32544_32x32_4',
  'Mshtml32545_32x32_4',
  'Shdocvw261_32x32_4',
  'Shdocvw274_32x32_4',
  'Shdocvw275_32x32_4',
];

/** Returns true if icon is not marked as blocked */
export const filterBlockedIcons = (icon: IconData): boolean =>
  !iconBlockedIds.includes(icon.id);
