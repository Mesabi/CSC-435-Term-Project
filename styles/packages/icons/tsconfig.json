{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "allowJs": true,
    "outDir": "dist/@types",
    "jsx": "react",
    "incremental": true,
    "declaration": true,
    "emitDeclarationOnly": true
  },
  "include": ["src"]
}
