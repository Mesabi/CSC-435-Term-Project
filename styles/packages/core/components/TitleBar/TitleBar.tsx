import React from 'react';
import styled, { th } from '@xstyled/styled-components';

import Button from '../Button';
import Frame, { FrameProps } from '../Frame/Frame';

const OptionsBox = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;

  display: flex;
`;

const Option = styled(Button).attrs({
  as: 'li',
})`
  display: flex;
  align-items: center;
  justify-content: center;

  margin-left: 2;
  padding: 0;

  width: 17px;
  height: 14px;

  min-width: 0;
  font-weight: 600;
  font-size: 10;

  &:active {
    padding: 1 0 0 1;

    outline: none;
  }

  &:first-child {
    margin-right: 0;
  }

  &:focus {
    box-shadow: inset 1px 1px 0px 1px ${th('colors.borderLightest')},
      inset -1px -1px 0px 1px ${th('colors.borderDark')};
  }
`;

Option.displayName = 'Option';

interface TitleBarBackgroundProps
  extends Omit<React.HTMLAttributes<HTMLDivElement>, 'color'>,
    FrameProps {
  active?: boolean;
  icon?: React.ReactElement;
  title?: string;
}

const TitleBarBackground = styled(Frame)<{ active?: boolean }>`
  height: 18;
  margin-bottom: 2;

  padding: 2;

  display: flex;

  box-shadow: none;

  img {
    width: 15;
    height: 14;
    margin-right: 4;
  }

  background: ${({ active, theme }) =>
    active
      ? theme.colors.headerBackground
      : theme.colors.headerNotActiveBackground};
`;

const Title = styled.div`
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
`;

interface TitleBarOptions {
  Option: typeof Option;
  OptionsBox: typeof OptionsBox;
}

export interface ITitleBarProps
  extends TitleBarBackgroundProps,
    TitleBarOptions {}

const TitleBarRenderer: React.ForwardRefRenderFunction<
  HTMLDivElement,
  TitleBarBackgroundProps
> = (
  { children, title = 'UNKNOWN.EXE', icon, active = true, className },
  ref: React.Ref<HTMLDivElement>,
) => (
  <TitleBarBackground
    active={active}
    color={active ? 'headerText' : 'headerNotActiveText'}
    ref={ref}
  >
    {icon && icon}
    <Title className={className}>{title}</Title>

    {children}
  </TitleBarBackground>
);

const TitleBar = Object.assign(
  React.forwardRef<HTMLDivElement, TitleBarBackgroundProps>(TitleBarRenderer),
  {
    Option,
    OptionsBox,
  },
) as ITitleBarProps & typeof TitleBarRenderer;

export default TitleBar;
