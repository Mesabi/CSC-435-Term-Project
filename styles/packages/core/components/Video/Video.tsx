import * as React from 'react';
import styled, { css } from '@xstyled/styled-components';
import { Mplayer113, User4 } from '@react95/icons';

import Frame, { FrameProps } from '../Frame/Frame';
import Button from '../Button';
import Range from '../Range';
import TitleBar from '../TitleBar';
import { Play, Pause, Stop, Fullscreen } from './buttons';
import Divider from '../List/ListDivider';

const VideoTag = styled.video<{ visible: boolean }>`
  width: 100%;
  padding: 2;

  display: ${({ visible }) => (visible ? 'block' : 'none')};
`;

type SourceProps = Pick<HTMLSourceElement, 'src'>;

const Source: React.FC<SourceProps> = ({ src }) => (
  <source src={src} type={`video/${src.substring(src.length - 3)}`} />
);

const ControlBtn = styled(Button)`
  display: inline-flex;
  justify-content: center;
  align-items: center;

  svg {
    fill: materialText;
  }

  &&,
  &:active,
  &:focus {
    width: 20px;
    height: 20px;
    padding: 7;

    ${({ disabled }) =>
      disabled &&
      css`
        padding: 4;
        svg {
          fill: borderDark;
          border-bottom: 1px solid;
          border-bottom-color: borderLightest;
          border-right: 1px solid;
          border-right-color: borderLightest;
        }
      `}
  }
`;

const Controls = styled.div`
  display: flex;
  align-items: center;
  padding: 2 0;
`;

const CountDownContainer = styled(Frame)`
  display: flex;
  padding: 6;
  margin-bottom: 4;

  box-shadow: in;
  background-color: canvas;
  height: 50px;

  color: canvasText;
`;

const VideoFont = styled.span`
  font-family: 'React95Video-Numbers';
  text-transform: uppercase;
`;

const ResetFrame = styled(Frame)`
  background-color: transparent;
  box-shadow: none;
`;

const VideoRange = styled(Range)`
  &::-webkit-slider-thumb {
    height: 18px;
    margin-top: -7px;
    width: 10px;
  }

  &[value='0']::-webkit-slider-thumb {
    margin-left: -2px;
  }
`;

const PlayOrPause = ({ playing }: { playing: boolean }) =>
  playing ? <Pause /> : <Play />;

const arrayFy = (str: string | string[]) => ([] as string[]).concat(str);

function updateProgressBar(
  player: HTMLVideoElement,
  updateProgress: (value: number) => void,
) {
  const percentage = Math.floor((100 / player.duration) * player.currentTime);

  updateProgress(percentage);
}

function parseCurrentTime(secs: number): string {
  if (!secs) {
    return '00:00';
  }

  const sec = parseInt(secs.toString(), 10);
  const hours = Math.floor(sec / 3600);
  const minutes = Math.floor(sec / 60) % 60;
  const seconds = sec % 60;

  return [hours, minutes, seconds]
    .map(v => (v < 10 ? `0${v}` : v))
    .filter((v, i) => v !== '00' || i > 0)
    .join(':');
}

export type VideoProps = {
  name?: string;
  src: string;
  videoProps?: React.HTMLAttributes<HTMLVideoElement>;
  style?: React.CSSProperties;
} & FrameProps;

export type VideoRefs = {
  video: React.Ref<HTMLVideoElement>;
  progress: React.Ref<HTMLInputElement>;
  wrapper: React.Ref<HTMLDivElement>;
  playpause: React.Ref<HTMLButtonElement>;
  stop: React.Ref<HTMLButtonElement>;
  fullScreen: React.Ref<HTMLButtonElement>;
};

const VideoRenderer = (
  { name, src, videoProps, style, ...props }: VideoProps,
  ref: React.Ref<VideoRefs>,
) => {
  const [playing, setPlaying] = React.useState(false);
  const [loadeddata, setLoadeddata] = React.useState(false);
  const [progress, setProgress] = React.useState(0);

  const player = React.useRef<HTMLVideoElement>(null);
  const progressRef = React.useRef<HTMLInputElement>(null);
  const wrapperRef = React.useRef<HTMLDivElement>(null);
  const playPauseRef = React.useRef<HTMLButtonElement>(null);
  const stopRef = React.useRef<HTMLButtonElement>(null);
  const fullScreenRef = React.useRef<HTMLButtonElement>(null);

  const paths = arrayFy(src);
  const [pathname] = paths;

  const normalizedName = name || pathname.replace(/^.*[\\/]/, '');
  const title = `${normalizedName}${!loadeddata ? ' (Opening)' : ''}`;

  React.useImperativeHandle(ref, () => ({
    get video() {
      return player;
    },
    get progress() {
      return progressRef;
    },
    get wrapper() {
      return wrapperRef;
    },
    get playpause() {
      return playPauseRef;
    },
    get stop() {
      return stopRef;
    },
    get fullScreen() {
      return fullScreenRef;
    },
  }));

  React.useEffect(() => {
    player.current?.addEventListener(
      'ended',
      () => {
        setPlaying(false);
        setProgress(0);
      },
      false,
    );

    player.current?.addEventListener(
      'timeupdate',
      () => {
        if (player.current) {
          updateProgressBar(player.current, setProgress);
        }
      },
      false,
    );

    player.current?.addEventListener(
      'loadeddata',
      () => {
        setLoadeddata(true);
      },
      false,
    );

    player.current?.addEventListener(
      'playing',
      () => {
        setPlaying(true);
      },
      false,
    );
  }, [player.current]);

  return (
    <Frame
      p={2}
      {...(props as typeof Frame as object)}
      style={{
        width: !loadeddata ? 260 : undefined,
        ...style,
      }}
      ref={wrapperRef}
    >
      <TitleBar icon={<Mplayer113 variant="16x16_4" />} title={title} />
      <VideoTag {...videoProps} visible={loadeddata} ref={player}>
        {paths.map(s => (
          <Source key={s} src={s} />
        ))}
      </VideoTag>
      {loadeddata && (
        <Divider as="span" style={{ display: 'block', marginBottom: 2 }} />
      )}
      <ResetFrame maxWidth="250" mx="auto" mb={4}>
        <CountDownContainer>
          <ResetFrame display="flex" flexDirection="column" w="40%">
            <VideoFont
              style={{
                marginTop: 'auto',
              }}
            >
              {player.current && parseCurrentTime(player.current.duration)}
            </VideoFont>

            <VideoFont style={{ height: 12 }}>
              {!loadeddata && 'Openning'}
            </VideoFont>
          </ResetFrame>
          <ResetFrame display="flex" flexDirection="column" w="40%">
            <VideoFont
              style={{
                marginTop: 'auto',
                fontSize: 22,
              }}
            >
              {player.current && parseCurrentTime(player.current.currentTime)}
            </VideoFont>

            <VideoFont style={{ height: 12 }}>time</VideoFont>
          </ResetFrame>
        </CountDownContainer>
        <Controls>
          <ControlBtn
            disabled={!loadeddata}
            onClick={() => {
              if (!playing) {
                player.current?.play();
              } else {
                player.current?.pause();
              }
              setPlaying(!playing);
            }}
            ref={playPauseRef}
          >
            {loadeddata ? (
              <PlayOrPause playing={playing} />
            ) : (
              <User4 variant="32x32_4" />
            )}
          </ControlBtn>
          <ControlBtn
            disabled={!loadeddata}
            onClick={() => {
              if (player.current) {
                player.current.pause();
                player.current.currentTime = 0;
              }

              setPlaying(false);
            }}
            ref={stopRef}
          >
            <Stop />
          </ControlBtn>
          <ControlBtn
            disabled={!loadeddata}
            onClick={() => {
              player?.current?.requestFullscreen();
            }}
            ref={fullScreenRef}
          >
            <Fullscreen />
          </ControlBtn>

          <VideoRange
            ref={progressRef}
            min="0"
            max="100"
            step="1"
            value={progress}
            style={{
              width: '70%',
              marginLeft: 20,
            }}
            onClick={e => {
              const { current: el } = progressRef;
              const { current: video } = player;

              if (video && el) {
                const percent = e.nativeEvent.offsetX / el.offsetWidth;

                video.currentTime = percent * video.duration;

                setProgress(Math.floor(percent / 100));
              }
            }}
          />
        </Controls>
      </ResetFrame>
    </Frame>
  );
};

const Video = React.forwardRef<VideoRefs, VideoProps>(VideoRenderer);

export default Video;
