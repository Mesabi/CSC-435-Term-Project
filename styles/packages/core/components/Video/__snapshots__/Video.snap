// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video /> Snapshot should match snapshot 1`] = `
.c12 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  color: #000000;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e, 1px 1px 0 0px #000000;
}

.c12:disabled {
  color: #868a8e;
}

.c12:focus {
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 1px 1px 0px 1px #ffffff, inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000000;
}

.c12:active {
  padding: 8px 20px 4px;
  outline: 1px dotted #000000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000000;
}

.c0 {
  padding: 2px;
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c1 {
  color: #fefefe;
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c5 {
  max-width: 250px;
  margin-right: auto;
  margin-left: auto;
  margin-bottom: 4px;
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c7 {
  width: auto;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 40%;
  height: auto;
  background-color: #c3c7cb;
  --x-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  box-shadow: var(--x-ring-shadow,0 0 #0000),var(--x-shadow);
}

.c2 {
  height: 18px;
  margin-bottom: 2px;
  padding: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: none;
  background: #000e7a;
}

.c2 img {
  width: 15px;
  height: 14px;
  margin-right: 4px;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  line-height: 1.4em;
  margin: 0;
  font-size: 1em;
}

.c14 {
  -webkit-appearance: none;
  width: 100%;
}

.c14:hover,
.c14:focus,
.c14:active {
  outline: none;
}

.c14::-webkit-slider-runnable-track {
  width: 100%;
  height: 4px;
  background-color: #000000;
  cursor: pointer;
  border-top: 1px solid;
  border-top-color: #d2d2d2;
  border-right: 2px solid;
  border-right-color: #ffffff;
  border-bottom: 1px solid;
  border-bottom-color: #e6e6e6;
  border-left: 2px solid;
  border-left-color: #ffffff;
  box-shadow: 0px 1px 0px 0px #ffffff;
}

.c14::-webkit-slider-thumb {
  width: 12px;
  height: 20px;
  background-color: #c3c7cb;
  cursor: pointer;
  margin-top: -8px;
  border-top: 1px solid;
  border-top-color: #ffffff;
  border-right: 1px solid;
  border-right-color: #000000;
  border-bottom: 1px solid;
  border-bottom-color: #000000;
  border-left: 1px solid;
  border-left-color: #ffffff;
  box-shadow: inset 0px -1px 0px #868a8e, inset -1px 0px 0px #868a8e, inset 0px 1px 0px #e6e6e6, inset 1px 0px 0px #e6e6e6;
  -webkit-appearance: none;
}

.c14:focus::-webkit-slider-runnable-track {
  background-color: #000000;
}

.c14::-moz-range-track {
  width: 100%;
  height: 4px;
  background-color: #000000;
  cursor: pointer;
  border-top: 1px solid;
  border-top-color: #d2d2d2;
  border-right: 2px solid;
  border-right-color: #ffffff;
  border-bottom: 1px solid;
  border-bottom-color: #e6e6e6;
  border-left: 2px solid;
  border-left-color: #ffffff;
  box-shadow: 0px 1px 0px 0px #ffffff;
}

.c14::-moz-range-thumb {
  width: 12px;
  height: 20px;
  background-color: #c3c7cb;
  cursor: pointer;
  margin-top: -8px;
  border-top: 1px solid;
  border-top-color: #ffffff;
  border-right: 1px solid;
  border-right-color: #000000;
  border-bottom: 1px solid;
  border-bottom-color: #000000;
  border-left: 1px solid;
  border-left-color: #ffffff;
  box-shadow: inset 0px -1px 0px #868a8e, inset -1px 0px 0px #868a8e, inset 0px 1px 0px #e6e6e6, inset 1px 0px 0px #e6e6e6;
  -webkit-appearance: none;
}

.c4 {
  width: 100%;
  padding: 2px;
  display: none;
}

.c13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 svg {
  fill: #000000;
}

.c13.c13,
.c13:active,
.c13:focus {
  width: 20px;
  height: 20px;
  padding: 7px;
  padding: 4px;
}

.c13.c13 svg,
.c13:active svg,
.c13:focus svg {
  fill: #868a8e;
  border-bottom: 1px solid;
  border-bottom-color: #ffffff;
  border-right: 1px solid;
  border-right-color: #ffffff;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2px 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 6px;
  margin-bottom: 4px;
  box-shadow: inset 0px 0px 0px 0px, inset 1px 1px 0px 0px #868a8e, 0.5px 0.5px 0px 0.5px #ffffff;
  background-color: #000000;
  height: 50px;
  color: #ffffff;
}

.c10 {
  font-family: 'React95Video-Numbers';
  text-transform: uppercase;
}

.c6 {
  background-color: transparent;
  box-shadow: none;
}

.c15::-webkit-slider-thumb {
  height: 18px;
  margin-top: -7px;
  width: 10px;
}

.c15[value='0']::-webkit-slider-thumb {
  margin-left: -2px;
}

<div>
  <div
    class="c0"
    style="width: 260px;"
  >
    <div
      class="c1 c2"
      color="headerText"
    >
      <img
        height="16"
        src="test-file-stub"
        width="16"
      />
      <div
        class="c3"
      >
        some_video.mp4 (Opening)
      </div>
    </div>
    <video
      class="c4"
    >
      <source
        src="foo/bar/some_video.mp4"
        type="video/mp4"
      />
    </video>
    <div
      class="c5 c6"
    >
      <div
        class="c7 c8"
      >
        <div
          class="c9 c6"
          display="flex"
        >
          <span
            class="c10"
            style="margin-top: auto;"
          />
          <span
            class="c10"
            style="height: 12px;"
          >
            Openning
          </span>
        </div>
        <div
          class="c9 c6"
          display="flex"
        >
          <span
            class="c10"
            style="margin-top: auto; font-size: 22px;"
          />
          <span
            class="c10"
            style="height: 12px;"
          >
            time
          </span>
        </div>
      </div>
      <div
        class="c11"
      >
        <button
          class="c12 c13"
          disabled=""
        >
          <img
            height="32"
            src="test-file-stub"
            width="32"
          />
        </button>
        <button
          class="c12 c13"
          disabled=""
        >
          <svg
            aria-label="stop"
            height="6"
            width="6"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h306v306H0z"
            />
          </svg>
        </button>
        <button
          class="c12 c13"
          disabled=""
        >
          <svg
            aria-label="fullscreen"
            height="6"
            viewBox="0 0 24 24"
            width="6"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 9h-2v-5h-7v-2h9v7zm-9 13v-2h7v-5h2v7h-9zm-15-7h2v5h7v2h-9v-7zm9-13v2h-7v5h-2v-7h9zm11 4h-16v12h16v-12z"
              style="width: 1000px;"
            />
          </svg>
        </button>
        <input
          class="c14 c15"
          max="100"
          min="0"
          step="1"
          style="width: 70%; margin-left: 20px;"
          type="range"
          value="0"
        />
      </div>
    </div>
  </div>
</div>
`;
